src/algorithms/model.ts

```diff
@@ -140,207 +144,261 @@ export function initializePopulation(
 
     return {
       time: t0,
-      susceptible: put(N - numCases),
-      exposed: putarr(0),
-      infectious: put(numCases),
-      hospitalized: put(0),
-      critical: put(0),
-      discharged: put(0),
-      recovered: put(0),
-      intensive: put(0),
-      dead: put(0),
-      overflow: put(0),
+      current: {
+        susceptible: put(N - numCases),
+        exposed: putarr(0),
+        infectious: put(numCases),
+        severe: put(0),
+        critical: put(0),
+        overflow: put(0),
+      },
+      cumulative: {
+        recovered: put(0),
+        hospitalized: put(0),
+        critical: put(0),
+        fatality: put(0),
+      },
     }
   }
   const Z = Object.values(ages).reduce((a, b) => a + b)
   const pop: SimulationTimePoint = {
     time: t0,
-    susceptible: {},
-    exposed: {},
-    infectious: {},
-    hospitalized: {},
-    critical: {},
-    overflow: {},
-    discharged: {},
-    intensive: {},
-    recovered: {},
-    dead: {},
+    current: {
+      susceptible: {},
+      exposed: {},
+      infectious: {},
+      severe: {},
+      critical: {},
+      overflow: {},
+    },
+    cumulative: {
+      recovered: {},
+      hospitalized: {},
+      critical: {},
+      fatality: {},
+    },
   }
   // TODO: Ensure the sum is equal to N!
   Object.keys(ages).forEach((k, i) => {
     const n = Math.round((ages[k] / Z) * N)
-    pop.susceptible[k] = n
-    pop.exposed[k] = [0, 0, 0]
-    pop.infectious[k] = 0
-    pop.hospitalized[k] = 0
-    pop.critical[k] = 0
-    pop.overflow[k] = 0
-    pop.discharged[k] = 0
-    pop.recovered[k] = 0
-    pop.intensive[k] = 0
-    pop.dead[k] = 0
+    pop.current.susceptible[k] = n
+    pop.current.exposed[k] = [0, 0, 0]
+    pop.current.infectious[k] = 0
+    pop.current.severe[k] = 0
+    pop.current.critical[k] = 0
+    pop.current.overflow[k] = 0
+    pop.cumulative.hospitalized[k] = 0
+    pop.cumulative.recovered[k] = 0
+    pop.cumulative.critical[k] = 0
+    pop.cumulative.fatality[k] = 0
     if (i === Math.round(Object.keys(ages).length / 2)) {
-      pop.susceptible[k] -= numCases
-      pop.infectious[k] = (0.3 * numCases)
-      const e           = (0.7 * numCases) / pop.exposed[k].length;
-      pop.exposed[k]    = [e, e, e];
+      pop.current.susceptible[k] -= numCases
+      pop.current.infectious[k] = 0.3 * numCases
+      const e = (0.7 * numCases) / pop.current.exposed[k].length
+      pop.current.exposed[k] = [e, e, e]
```

Comment:
if we can make this not imply the length of the array, that would be great!

Reply:
Updated the branch. The value is still hardcoded in one spot  but I think this is OK as everything else will follow from that.